ofp/match.go

```diff
@@ -306,14 +305,15 @@ func (xm *XM) ReadFrom(r io.Reader) (n int64, err error) {
 	hasmask := (xm.Type & 1) == 1
 	xm.Type >>= 1
 
-	var m int64
-
 	xm.Value, xm.Mask = make(XMValue, length), nil
-	m, err = encoding.ReadFrom(r, &xm.Value)
-	n += m
 
-	if err != nil {
-		return
+	if hasPayload {
```

Comment:
Is it possible to determine if OXM has payload by analyzing its length? So the caller won't need to pas ```hasPayload``` flag?

I mean precisely this:
```go
if length != 0 {
    m, err := encoding.ReadFrom(r, &xm.Value)
    n += m
    if err != nil {
        return err
    }
}
```

Reply:
It's not possible.  The Length when `hasPayload == false` indicates the length that a hypothetical payload for the given XM type should be.